optaweb-vehicle-routing-backend/src/main/java/org/optaweb/vehiclerouting/plugin/planner/SolverManager.java

```diff
@@ -96,22 +101,30 @@ void startSolver(VehicleRoutingSolution solution) {
         if (solverFuture != null) {
             throw new IllegalStateException("Solver start has already been requested");
         }
-        solverFuture = executor.submitListenable((SolvingTask) () -> solver.solve(solution));
-        solverFuture.addCallback(
+        solverFuture = executor.submit((SolvingTask) () -> solver.solve(solution));
+        solverFuture.addListener(
                 // IMPORTANT: This is happening on the solver thread.
-                // TODO in both cases restart or somehow recover?
-                result -> {
+                // TODO maybe restart or somehow recover?
+                () -> {
                     if (!solver.isTerminateEarly()) {
-                        // This is impossible. Solver in daemon mode can't return from solve() unless it has been
-                        // terminated (see #stopSolver()) or throws an exception.
-                        logger.error("Solver stopped solving but that shouldn't happen in daemon mode.");
-                        eventPublisher.publishEvent(new ErrorEvent(this, "Solver stopped solving unexpectedly."));
+                        // Solver in daemon mode can't return from solve() unless it has been terminated early
+                        // (see #stopSolver()).
+                        // So this case is only possible when an exception is thrown during solver.solve().
+                        try {
+                            solverFuture.get();
+                            logger.error("This should be impossible. The solver has stopped without being terminated early"
```

Comment:
Could you please assure me - when this would happen the user will definitely have his logger switched on error level to be able to see that

Reply:
Yes, error level is enabled by default and the message is logged and visible on the standard output. In addition to that, an error event is fired and propagated to the UI.